fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java

```diff
@@ -717,50 +741,60 @@ private boolean isAllowed(Resource resource, Set<String> contextIds, Permission
+
+                    // In the case of search, Provenance resources can be in the response bundle for two reasons:
+                    // 1. direct search
+                    // 2. _revinclude from a different resource type
+                    // For case 1, the search is already scoped to the patient compartment and therefore only approved resources should be included
+                    // For case 2, only Provenance resources which target to another resource in the response bundle will be included and therefor we
+                    // can base the access decision on those resources rather than the Provenance
+
+                    // In the case of read/vread/history, access to Provenance will be handled elsewhere;
+                    // not by its Patient compartment membership but by the membership of the resources which it targets
                     if (log.isLoggable(Level.FINE)) {
-                        log.fine(requiredPermission.value() + " permission for 'List/" + resource.getId() +
+                        log.fine(requiredPermission.value() + " permission for 'Provenance/" + resource.getId() +
                             "' is granted via scope " + approvedScopeMap.get(ContextType.PATIENT));
                     }
                     return true;
+                } else if (resource instanceof com.ibm.fhir.model.resource.List) {
+                    // List is another special-case:  a Patient-compartment resource type that
+                    // we may want to grant access to when it is not scoped to any particular patient (e.g. for
+                    // <a href="http://hl7.org/fhir/us/davinci-drug-formulary/StructureDefinition-usdf-CoveragePlan.html">Drug Formulary</a>).
+
+                    // TODO: consider double-checking that the Patient compartment inclusion criteria for List have no
+                    // patient references.  The inclusion criteria are "subject" and "source".
+                    if (resource.getMeta() != null &&
+                            resource.getMeta().getProfile().contains(Canonical.of(DAVINCI_DRUG_FORMULARY_COVERAGE_PLAN))) {
+                        if (log.isLoggable(Level.FINE)) {
+                            log.fine(requiredPermission.value() + " permission for 'List/" + resource.getId() +
+                                "' is granted via scope " + approvedScopeMap.get(ContextType.PATIENT));
+                        }
+                        return true;
+                    }
                 }
-            }
 
-            // Else, see if the target resource belongs to the Patient compartment of the in-context patient
-            return isInCompartment(resource, CompartmentType.PATIENT, contextIds);
+                // Else, see if the target resource belongs to the Patient compartment of the in-context patient
+                return isInCompartment(resource, CompartmentType.PATIENT, contextIds);
+            } else {
```

Comment:
style-only comment:  personally i prefer the smaller block first so that you don't have to remember what the else on line 789 was lined up with above

it could even be made as a guard clause with its own early exit:
```
if (resource == null) {
  log.fine(...);
  return false;
}
```

then you could remove one level of indentation

Reply:
although i suppose some folks prefer a single exit point of the function at the bottom (e.g.in case someone adds some other type of checking after the `approvedScopeMap.containsKey(ContextType.PATIENT` block

Reply:
Yeah, you exactly followed my thought process.  I originally had the early exit, and then thought about if someone wanted to add more logic after it.

Reply:
Updated.

Reply:
thanks.  even without the early exit, i think this version is a lot easier for my brain